--- a/configure
+++ b/configure
@@ -36,7 +36,7 @@ OPTIONS=(
   "avahi:auto"
   "zlib:auto"
   "libav:auto"
-  "ffmpeg_static:yes"
+  "ffmpeg_static:no"
   "libx264:yes"
   "libx264_static:yes"
   "libx265:yes"
--- a/Makefile
+++ b/Makefile
@@ -52,7 +52,7 @@ CFLAGS  += -Wall -Wwrite-strings -Wno-de
 CFLAGS  += -Wmissing-prototypes
 CFLAGS  += -fms-extensions -funsigned-char -fno-strict-aliasing
 CFLAGS  += -D_FILE_OFFSET_BITS=64
-CFLAGS  += -I${BUILDDIR} -I${ROOTDIR}/src -I${ROOTDIR}
+CFLAGS  += -I ./vendor/dvb-api/5.10 -I${BUILDDIR} -I${ROOTDIR}/src -I${ROOTDIR}
 ifeq ($(CONFIG_ANDROID),yes)
 LDFLAGS += -ldl -lm
 else
