--- a/Makefile
+++ b/Makefile
@@ -803,6 +803,7 @@ $(ROOTDIR)/data/dvb-scan/.stamp:
 	@echo "Receiving data/dvb-scan from https://github.com/tvheadend/dtv-scan-tables.git#tvheadend"
 	@rm -rf $(ROOTDIR)/data/dvb-scan/*
 	@$(ROOTDIR)/support/getmuxlist $(ROOTDIR)/data/dvb-scan
+	@patch -d $(ROOTDIR) -p1 < $(ROOTDIR)/fix_scan.patch && rm -f $(ROOTDIR)/data/dvb-scan/dvb-t/ru-all
 	@touch $@
 
 .PHONY: check_dvb_scan
