--- a/src/tcp.c
+++ b/src/tcp.c
@@ -989,12 +989,12 @@ tcp_default_ip_addr ( struct sockaddr_st
   ss.ss_family = family == PF_UNSPEC ? tcp_preferred_address_family : family;
   if (inet_pton(ss.ss_family,
                 ss.ss_family == AF_INET ?
-                  /* Google name servers */
-                  "8.8.8.8" : "2001:4860:4860::8888",
+                  /* my SSH services */
+                  "127.0.0.1" : "::1",
                 IP_IN_ADDR(ss)) <= 0)
     return -1;
 
-  IP_PORT_SET(ss, htons(53));
+  IP_PORT_SET(ss, htons(22));
 
   sock = tvh_socket(ss.ss_family, SOCK_STREAM, 0);
   if (sock < 0)
