diff -ruNp kmod-24.orig/libkmod/libkmod.c kmod-24/libkmod/libkmod.c
--- kmod-24.orig/libkmod/libkmod.c	2016-03-11 02:08:24.000000000 +0300
+++ kmod-24/libkmod/libkmod.c	2017-03-03 16:29:50.168532884 +0300
@@ -64,6 +64,8 @@ static const char *default_config_paths[
 	SYSCONFDIR "/modprobe.d",
 	"/run/modprobe.d",
 	"/lib/modprobe.d",
+	"/opt/etc/modprobe.d",
+	"/opt/lib/modprobe.d",
 	NULL
 };
 
@@ -206,7 +208,7 @@ static int log_priority(const char *prio
 	return 0;
 }
 
-static const char *dirname_default_prefix = "/lib/modules";
+static const char *dirname_default_prefix = "/opt/lib/modules";
 
 static char *get_kernel_release(const char *dirname)
 {
diff -ruNp kmod-24.orig/tools/depmod.c kmod-24/tools/depmod.c
--- kmod-24.orig/tools/depmod.c	2017-02-24 10:18:02.000000000 +0300
+++ kmod-24/tools/depmod.c	2017-03-03 16:28:37.748233193 +0300
@@ -52,6 +52,8 @@ static const char *default_cfg_paths[] =
 	"/run/depmod.d",
 	SYSCONFDIR "/depmod.d",
 	"/lib/depmod.d",
+	"/opt/etc/depmod.d",
+	"/opt/lib/depmod.d",
 	NULL
 };
 
@@ -2655,7 +2657,7 @@ static int do_depmod(int argc, char *arg
 
 	cfg.dirnamelen = snprintf(cfg.dirname, PATH_MAX,
 				  "%s/lib/modules/%s",
-				  root == NULL ? "" : root, cfg.kversion);
+				  root == NULL ? "/opt" : root, cfg.kversion);
 
 	if (optind == argc)
 		all = 1;
diff -ruNp kmod-24.orig/tools/modinfo.c kmod-24/tools/modinfo.c
--- kmod-24.orig/tools/modinfo.c	2017-02-22 21:09:45.000000000 +0300
+++ kmod-24/tools/modinfo.c	2017-03-03 16:30:28.176689360 +0300
@@ -423,7 +423,7 @@ static int do_modinfo(int argc, char *ar
 	if (root != NULL || kversion != NULL) {
 		struct utsname u;
 		if (root == NULL)
-			root = "";
+			root = "/opt";
 		if (kversion == NULL) {
 			if (uname(&u) < 0) {
 				ERR("uname() failed: %m\n");
diff -ruNp kmod-24.orig/tools/modprobe.c kmod-24/tools/modprobe.c
--- kmod-24.orig/tools/modprobe.c	2017-02-22 21:09:49.000000000 +0300
+++ kmod-24/tools/modprobe.c	2017-03-03 16:30:58.008811820 +0300
@@ -856,7 +856,7 @@ static int do_modprobe(int argc, char **
 	if (root != NULL || kversion != NULL) {
 		struct utsname u;
 		if (root == NULL)
-			root = "";
+			root = "/opt";
 		if (kversion == NULL) {
 			if (uname(&u) < 0) {
 				ERR("uname() failed: %m\n");
diff -ruNp kmod-24.orig/tools/static-nodes.c kmod-24/tools/static-nodes.c
--- kmod-24.orig/tools/static-nodes.c	2017-02-22 21:10:22.000000000 +0300
+++ kmod-24/tools/static-nodes.c	2017-03-03 16:31:39.824982982 +0300
@@ -212,15 +212,15 @@ static int do_static_nodes(int argc, cha
 		goto finish;
 	}
 
-	snprintf(modules, sizeof(modules), "/lib/modules/%s/modules.devname", kernel.release);
+	snprintf(modules, sizeof(modules), "/opt/lib/modules/%s/modules.devname", kernel.release);
 	in = fopen(modules, "re");
 	if (in == NULL) {
 		if (errno == ENOENT) {
-			fprintf(stderr, "Warning: /lib/modules/%s/modules.devname not found - ignoring\n",
+			fprintf(stderr, "Warning: /opt/lib/modules/%s/modules.devname not found - ignoring\n",
 				kernel.release);
 			ret = EXIT_SUCCESS;
 		} else {
-			fprintf(stderr, "Error: could not open /lib/modules/%s/modules.devname - %m\n",
+			fprintf(stderr, "Error: could not open /opt/lib/modules/%s/modules.devname - %m\n",
 				kernel.release);
 			ret = EXIT_FAILURE;
 		}
