--- a/libkmod/libkmod.c
+++ b/libkmod/libkmod.c
@@ -66,6 +66,8 @@ static const char *default_config_paths[
 	"/run/modprobe.d",
 	"/usr/local/lib/modprobe.d",
 	"/lib/modprobe.d",
+	"/opt/etc/modprobe.d",
+	"/opt/lib/modprobe.d",
 	NULL
 };
 
@@ -208,7 +210,7 @@ static int log_priority(const char *prio
 	return 0;
 }
 
-static const char *dirname_default_prefix = "/lib/modules";
+static const char *dirname_default_prefix = "/opt/lib/modules";
 
 static char *get_kernel_release(const char *dirname)
 {
--- a/tools/depmod.c
+++ b/tools/depmod.c
@@ -55,6 +55,8 @@ static const char *default_cfg_paths[] =
 	"/run/depmod.d",
 	"/usr/local/lib/depmod.d",
 	"/lib/depmod.d",
+	"/opt/etc/depmod.d",
+	"/opt/lib/depmod.d",
 	NULL
 };
 
@@ -3009,7 +3011,7 @@ static int do_depmod(int argc, char *arg
 
 	cfg.dirnamelen = snprintf(cfg.dirname, PATH_MAX,
 				  "%s/lib/modules/%s",
-				  root == NULL ? "" : root, cfg.kversion);
+				  root == NULL ? "/opt" : root, cfg.kversion);
 
 	if (optind == argc)
 		all = 1;
--- a/tools/modinfo.c
+++ b/tools/modinfo.c
@@ -454,7 +454,7 @@ static int do_modinfo(int argc, char *ar
 	if (root != NULL || kversion != NULL) {
 		struct utsname u;
 		if (root == NULL)
-			root = "";
+			root = "/opt";
 		if (kversion == NULL) {
 			if (uname(&u) < 0) {
 				ERR("uname() failed: %m\n");
--- a/tools/modprobe.c
+++ b/tools/modprobe.c
@@ -960,7 +960,7 @@ static int do_modprobe(int argc, char **
 	if (root != NULL || kversion != NULL) {
 		struct utsname u;
 		if (root == NULL)
-			root = "";
+			root = "/opt";
 		if (kversion == NULL) {
 			if (uname(&u) < 0) {
 				ERR("uname() failed: %m\n");
--- a/tools/static-nodes.c
+++ b/tools/static-nodes.c
@@ -212,15 +212,15 @@ static int do_static_nodes(int argc, cha
 		goto finish;
 	}
 
-	snprintf(modules, sizeof(modules), "/lib/modules/%s/modules.devname", kernel.release);
+	snprintf(modules, sizeof(modules), "/opt/lib/modules/%s/modules.devname", kernel.release);
 	in = fopen(modules, "re");
 	if (in == NULL) {
 		if (errno == ENOENT) {
-			fprintf(stderr, "Warning: /lib/modules/%s/modules.devname not found - ignoring\n",
+			fprintf(stderr, "Warning: /opt/lib/modules/%s/modules.devname not found - ignoring\n",
 				kernel.release);
 			ret = EXIT_SUCCESS;
 		} else {
-			fprintf(stderr, "Error: could not open /lib/modules/%s/modules.devname - %m\n",
+			fprintf(stderr, "Error: could not open /opt/lib/modules/%s/modules.devname - %m\n",
 				kernel.release);
 			ret = EXIT_FAILURE;
 		}
