--- a/configure
+++ b/configure
@@ -663,9 +663,9 @@ if enabled_or_auto tvhcsa; then
     enable tvhcsa
     enable dvbcsa
     if enabled dvbcsa; then
-      (check_cc_header "dvbcsa/dvbcsa" dvbcsa_h &&\
-       check_cc_lib    dvbcsa dvbcsa_l) ||\
-      die "Failed to find dvbcsa library"
+#      (check_cc_header "dvbcsa/dvbcsa" dvbcsa_h &&\
+#       check_cc_lib    dvbcsa dvbcsa_l) ||\
+#      die "Failed to find dvbcsa library"
       LDFLAGS="$LDFLAGS -ldvbcsa"
     fi
   else
