diff -ruNp trunk.orig/Makefile trunk/Makefile
--- trunk.orig/Makefile	2018-03-25 14:31:51.587353173 +0300
+++ trunk/Makefile	2018-03-25 14:56:11.591981095 +0300
@@ -5,7 +5,7 @@ SHELL = /bin/sh
 .PHONY: all tests help README.build README.config simple default debug config menuconfig allyesconfig allnoconfig defconfig clean distclean
 
 VER     := $(shell ./config.sh --oscam-version)
-SVN_REV := $(shell ./config.sh --oscam-revision)
+#SVN_REV := $(shell ./config.sh --oscam-revision)
 
 uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
 
@@ -35,7 +35,7 @@ ifdef USE_SSL
 	override USE_LIBCRYPTO=1
 endif
 
-CONF_DIR = /usr/local/etc
+CONF_DIR = /opt/etc/oscam
 
 LIB_PTHREAD = -lpthread
 LIB_DL = -ldl
