--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,11 @@
-CC=gcc
+CC ?=gcc
 
 SRC=atsc_psip_section.c diseqc.c dump-vdr.c dump-zap.c dump-m3u.c lnb.c scan.c section.c 
 HED=atsc_psip_section.h diseqc.h dump-vdr.h dump-zap.h dump-m3u.h lnb.h scan.h section.h list.h
 OBJ=atsc_psip_section.o diseqc.o dump-vdr.o dump-zap.o dump-m3u.o lnb.o scan.o section.o
 
-BIND=/usr/local/bin/
-INCLUDE=-I../s2/linux/include
+BIND=/opt/usr/bin/
+INCLUDE=-I./include
 
 TARGET=scan-s2
 
