--- a/dump-m3u.c
+++ b/dump-m3u.c
@@ -239,7 +239,7 @@ void m3u_dump_service_parameter_set (FIL
 
 	if(!start_header_m3u)
 	{
-		fprintf (f, "#EXTM3U\n");
+		fprintf (f, "#EXTM3U name=\"My DVB scan list\"\n");
 		start_header_m3u = 1;
 	}
 
--- a/scan.c
+++ b/scan.c
@@ -3141,7 +3141,7 @@ static const char *usage = "\n"
 "	-I cnt	Scan iterations count (default 10).\n"
 "		Larger number will make scan longer on every channel\n"
 "	-M	Scan with support Multiple-PLP (DVB-T2 only)\n"
-"	-H url	Generation M3U playlist for SATIP, use as 'http://host:port' or 'rtsp://host:port'\n"
+"	-H url	Generation M3U playlist for SATIP, use as 'http://host:port', 'rtsp://host[:port]' or 'satip://host[:port]'\n"
 "	-o fmt	output format: 'm3u', 'vdr' (default), 'vdr16x' for VDR version 1.6.x or 'zap'\n"
 "	-x N	Conditional Access, (default -1)\n"
 "		N=-2  gets all channels (FTA and encrypted),\n"
@@ -3300,7 +3300,7 @@ int main (int argc, char **argv)
 			break;
 
 		case 'H':
-			if(!strncmp(optarg, "http://", 7) || !strncmp(optarg, "rtsp://", 7))
+			if(!strncmp(optarg, "http://", 7) || !strncmp(optarg, "rtsp://", 7) || !strncmp(optarg, "satip://", 8))
 				strncpy(url, optarg, sizeof(url)-1);
 			else {
 				bad_usage(argv[0], 0);
