diff -ruN scan-s2.orig/Makefile scan-s2/Makefile
--- scan-s2.orig/Makefile	2017-03-29 01:36:43.000000000 +0300
+++ scan-s2/Makefile	2017-03-29 01:45:10.608118747 +0300
@@ -12,7 +12,7 @@
 all: $(TARGET)
 
 $(TARGET): $(OBJ)
-	$(CC) $(CFLG) $(OBJ) -o $(TARGET) $(CLIB) 
+	$(CC) $(CFLAGS) $(OBJ) -o $(TARGET) $(CLIB)
 
 $(OBJ): $(HED)
 
@@ -26,4 +26,4 @@
 	rm -f $(OBJ) $(TARGET) *~
 
 %.o: %.c
-	$(CC) $(INCLUDE) -c $< -o $@
+	$(CC) $(CFLAGS) $(INCLUDE) -c $< -o $@
