diff -ruN scan-s2.orig/dump-m3u.c scan-s2/dump-m3u.c
--- scan-s2.orig/dump-m3u.c	2017-03-29 02:11:20.000000000 +0300
+++ scan-s2/dump-m3u.c	2017-03-30 13:31:44.098465638 +0300
@@ -251,7 +251,14 @@
 
 		m3u_dvb_parameters (f, t, 1);
 
-		fprintf (f, "pids=0,%i,%i", s->service_id, s->audio_pid[0]);
+		fprintf (f, "pids=0");
+		if(!s->audio_pid[0] && !s->audio_num && !s->video_pid && !s->teletext_pid)
+		{
+			fprintf (f, ",all\n");
+			return;
+		}
+		fprintf (f, ",%i", s->service_id);
+		fprintf (f, ",%i", s->audio_pid[0]);
 		for (i = 1; i < s->audio_num; i++)
 			fprintf (f, ",%i", s->audio_pid[i]);
 		if(s->video_pid > 0)
